doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,175 @@
+C++ Plugin API for CPU compressors
+============
+
+This document specifies a modern API for the in-memory C plugin API. It
+will not replace the CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current CPU compression plugins
+    * Support 1D and 2D data layouts
+    * Drop alpha during compression of 2D data with alpha
+    * Lossy compression for 2D data
+    * Support for common image pixel layouts
+    * Support for different 1D data (byte, float, double, ...)
+    * Multiple compressors per plugin DSO
+* Information provided for each compressor:
+    * Compression speed (relative to RLE compressor)
+    * Compression ratio (relative to RLE compressor)
+    * 1D or 2D compressor
+    * Implemention drops alpha (for 2D)
+* Compressors may produce multiple output chunks
+* Compressor engines are instantiated to allow reuse of internal state
+* Easy to use API for plugin implementers
+
+## Dependency Changes
+
+* New: Lunchbox plugin API for loading plugins
+
+## API
+
+Change signature of Lunchbox::Plugin::HandlesFunc to:
+    typedef boost::function< bool ( InitDataT& ) > HandlesFunc;
```

Comment:
Instead of the plugins registering with the global factory, the factory instance already loads and queries the plugins, and can register them in this step. No global variable needed, but the existing C entry points need to be augmented.